cedar-policy-core/src/ast/entity.rs

```diff
@@ -25,47 +25,101 @@ use crate::FromNormalizedStr;
 use educe::Educe;
 use itertools::Itertools;
 use miette::Diagnostic;
-use ref_cast::RefCast;
-use serde::{Deserialize, Serialize};
+use serde::{de::Deserializer, ser::Serializer, Deserialize, Serialize};
 use serde_with::{serde_as, TryFromInto};
 use smol_str::SmolStr;
 use std::collections::{BTreeMap, HashMap, HashSet};
 use std::str::FromStr;
 use std::sync::Arc;
 use thiserror::Error;
 
+#[cfg(feature = "tolerant-ast")]
+static ERROR_NAME: std::sync::LazyLock<Name> =
+    std::sync::LazyLock::new(|| Name(InternalName::from(Id::new_unchecked("EntityTypeError"))));
+
+#[cfg(feature = "tolerant-ast")]
+static ERROR_EID_SMOL_STR: std::sync::LazyLock<SmolStr> =
+    std::sync::LazyLock::new(|| SmolStr::from("Eid::ErrorEid"));
```

Comment:
nit, but I wonder if a `std::sync::LazyLock` is worth it when all we're saving is the construction of one `SmolStr`.  (the alternative being `pub static ERROR_EID: &str` and just do `SmolStr::from` every time the `SmolStr` is needed)

Reply:
Ah never mind, actually the issue here is that we can't return a reference to a local variable. I think the simplest way is what is already here, I'll leave it